src/client/auth.ts

```
@@ -162,6 +165,15 @@ function selectClientAuthMethod(
     return hasClientSecret ? "client_secret_post" : "none";
   }
 
+  // Prefer the method returned by the server during client registration if valid and supported
+  if (
+    clientInformation.token_endpoint_auth_method &&
+    isClientAuthMethod(clientInformation.token_endpoint_auth_method) &&
+    supportedMethods.includes(clientInformation.token_endpoint_auth_method)
+  ) {
+    return clientInformation.token_endpoint_auth_method;
+  }
+
```

Should this throw an error if an invalid or unsupported `token_endpoint_auth_method` is returned? This implementation would simply skip it and fall through to the rest of the logic if it is invalid or unsupported.

Also, should it care if it's valid or supported in this context? or should it just use the `token_endpoint_auth_method` value without validating it at all?